### PR TITLE
RHAIENG-608: Update all runtimes imagestream names with new naming format

### PR DESCRIPTION
**Pull request by dibryant**
Fixes for https://issues.redhat.com/browse/RHAIENG-608
## Description

Update all runtimes imagestream names with new naming format
## How Has This Been Tested?

Self checklist (all need to be checked):
- [ ] Ensure that you have run `make test` (`gmake` on macOS) before asking for review
- [ ] Changes to everything except `Dockerfile.konflux` files should be done in `odh/notebooks` and automatically synced to `rhds/notebooks`. For Konflux-specific changes, modify `Dockerfile.konflux` files directly in `rhds/notebooks` as these require special attention in the downstream repository and flow to the upcoming RHOAI release.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

## Summary by CodeRabbit

* **Style**
  * Standardized display names for Python 3.12 runtimes to the "Runtime | <Type> | <Accelerator> | Python 3.12" format.
  * Updated runtime labels for Datascience (CPU), Minimal (CPU), PyTorch (CUDA), PyTorch (ROCm), TensorFlow (CUDA), TensorFlow (ROCm).
  * Improves consistency and clarity in runtime selection menus and image listings, making choices easier to scan and compare.

### CODE REVIEW

File: manifests/base/runtime-rocm-pytorch-imagestream.yaml [hunk 1]
```diff
@@ -18,12 +18,12 @@ spec:
         opendatahub.io/runtime-image-metadata: |
           [
             {
-              "display_name": "PyTorch with ROCm and Python 3.12 (UBI9)",
+              "display_name": "Runtime | Pytorch | ROCm| Python 3.12",
```
Comment (atheo89) [inline manifests/base/runtime-rocm-pytorch-imagestream.yaml:21]:
```suggestion
              "display_name": "Runtime | Pytorch | ROCm | Python 3.12",
```
this need to updated in several places

**Review by atheo89 — APPROVED** [review]
/lgtm
